shared/templates/file_groupowner/oval.template

```diff
@@ -38,11 +38,34 @@
   </unix:file_object>
 
   <unix:file_state id="state_file_groupowner{{{ FILEID }}}_gid_{{{ FILEGID }}}_{{{ loop.index0 }}}" version="1">
+    {{%- if FILEGID == '0' %}}
     <unix:group_id datatype="int">{{{ FILEGID }}}</unix:group_id>
+    {{%- else %}}
+    <unix:group_id datatype="int" var_ref="var_dedicated_groupowner{{{ FILEID }}}_uid_{{{ FILEGID }}}"></unix:group_id>
+    {{%- endif %}}
   </unix:file_state>
   {{% endfor %}}
 
   <unix:file_state id="symlink_file_groupowner{{{ FILEID }}}_uid_{{{ FILEGID }}}" version="1">
     <unix:type operation="equals">symbolic link</unix:type>
   </unix:file_state>
+
+  {{%- if FILEGID != '0' %}}
```

Comment:
A explanatory variable `gid_is_name` would fit perfectly here, and probably should be more robust - numbers are probably IDs, not numbers are definitely names. Moreover, I believe that the template code could propagate a boolean with the template.